src/spaces/jolt_contact_listener_3d.cpp

```diff
@@ -333,6 +337,55 @@ void JoltContactListener3D::flush_area_exits() {
 	area_exits.clear();
 }
 
+void JoltContactListener3D::override_collision_response(
+	const JPH::Body& p_body1,
+	const JPH::Body& p_body2,
+	JPH::ContactSettings& p_settings
+) {
+	if (p_body1.IsSensor() || p_body2.IsSensor()) {
+		return;
+	}
+
+	if (!p_body1.IsDynamic() && !p_body2.IsDynamic()) {
+		return;
+	}
+
+	JPH::BroadPhaseLayer broad_phase_layer1 = {};
+	uint32_t collision_layer1 = 0;
+	uint32_t collision_mask1 = 0;
+
+	space->map_from_object_layer(
+		p_body1.GetObjectLayer(),
+		broad_phase_layer1,
+		collision_layer1,
+		collision_mask1
+	);
+
+	JPH::BroadPhaseLayer broad_phase_layer2 = {};
+	uint32_t collision_layer2 = 0;
+	uint32_t collision_mask2 = 0;
+
+	space->map_from_object_layer(
+		p_body2.GetObjectLayer(),
+		broad_phase_layer2,
+		collision_layer2,
+		collision_mask2
+	);
+
+	const bool first_scans_second = (collision_mask1 & collision_layer2) != 0;
+	const bool second_scans_first = (collision_mask2 & collision_layer1) != 0;
+
+	if (p_body1.IsKinematic() || (second_scans_first && !first_scans_second)) {
+		p_settings.mInvMassScale1 = 0.0f;
+		p_settings.mInvInertiaScale1 = 0.0f;
+	}
+
+	if (p_body2.IsKinematic() || (first_scans_second && !second_scans_first)) {
+		p_settings.mInvMassScale2 = 0.0f;
+		p_settings.mInvInertiaScale2 = 0.0f;
+	}
```

Comment:
@jrouwe Needing to explicitly override the mass/inertia for kinematic bodies here struck me as a bit strange. Shouldn't they by default have that already?

If I leave those conditions out the collision ends up being somewhat strange, where the non-kinematic body (with infinite mass/inertia) will still collide with the kinematic body, but end up penetrating it ever so slightly before coming to a full stop.

Reply:
Here's a video of the scene from #400 showing the behavior when those `IsKinematic()` checks are omitted:

https://github.com/godot-jolt/godot-jolt/assets/4884246/24e42a3d-01ce-4e5a-9abc-9997ee6d0621

Reply:
Yes, I didn't test this properly. I knew that the value would be unused for a kinematic body, but it was actually used in 1 place :)

The fix is in https://github.com/jrouwe/JoltPhysics/tree/bugfix/kinematic_vs_overridden_mass

I still need to do a bit of testing on it

Reply:
Fix is submitted